diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2013-10-15 21:15:16.000000000 +0200
+++ b/configure.ac	2013-10-19 20:47:30.290093584 +0200
@@ -1504,7 +1504,7 @@
 # installed; if they're still doing that in current Fedora releases,
 # perhaps there will also be XXX-qt5 when they pick up Qt 5.
 #
-AC_PATH_PROG(UIC, uic)
+UIC="/usr/lib64/qt5/bin/uic"
 if test "x$UIC" = x
 then
 	AC_PATH_PROG(UIC, uic-qt4)
@@ -1530,7 +1530,7 @@
 	fi
 fi
 AC_SUBST(UIC)
-AC_PATH_PROG(MOC, moc)
+MOC="/usr/lib64/qt5/bin/moc"
 if test "x$MOC" = x
 then
 	AC_PATH_PROG(MOC, moc-qt4)
