diff --git a/deployment.pri b/deployment.pri
index 5441b63..81cde64 100644
--- a/deployment.pri
+++ b/deployment.pri
@@ -17,7 +17,7 @@ android-no-sdk {
         qnx {
             target.path = /tmp/$${TARGET}/bin
         } else {
-            target.path = /opt/$${TARGET}/bin
+            target.path = /usr/bin
         }
         export(target.path)
     }
diff --git a/mobile-datovka.pro b/mobile-datovka.pro
index be34edd..5e58130 100644
--- a/mobile-datovka.pro
+++ b/mobile-datovka.pro
@@ -1,12 +1,5 @@
 QT += qml quick quickcontrols2 network xml sql widgets svg printsupport
 
-android {
-	QT += androidextras
-	# Android NDK r17c does not need the '-nostdlib++' option. Starting
-	# from NDK r18 the tool chain uses Clang which needs this parameter.
-	QMAKE_LINK += -nostdlib++
-}
-
 TEMPLATE = app
 APP_NAME = Datovka
 # winphone|winrt target.
@@ -82,9 +75,9 @@ command = $$LRELEASE mobile-datovka.pro
 system($$command)|error("Failed to run: $$command")
 
 # Copy translation files at qmake invocation.
-message(Copying Qt translation from $$system_path($$[QT_INSTALL_DATA]).)
-system($$QMAKE_COPY $$system_path($$[QT_INSTALL_DATA]/translations/qtbase_cs.qm) $$system_path(res/locale/qtbase_cs.qm))
-system($$QMAKE_COPY $$system_path($$[QT_INSTALL_DATA]/translations/qtbase_en.qm) $$system_path(res/locale/qtbase_en.qm))
+##message(Copying Qt translation from $$system_path($$[QT_INSTALL_DATA]).)
+##system($$QMAKE_COPY $$system_path($$[QT_INSTALL_DATA]/translations/qtbase_cs.qm) $$system_path(res/locale/qtbase_cs.qm))
+##system($$QMAKE_COPY $$system_path($$[QT_INSTALL_DATA]/translations/qtbase_en.qm) $$system_path(res/locale/qtbase_en.qm))
 
 TRANSLATIONS_FILES += \
     res/locale/qtbase_cs.qm \
@@ -391,19 +384,9 @@ HEADERS += \
     src/xml/xml_send_message.h \
     src/zfo.h
 
-android {
-	SOURCES += \
-	    android/src/android_io.cpp \
-	    src/os_android.cpp
-
-	HEADERS += \
-	    android/src/android_io.h \
-	    src/os_android.h
-}
-
 RESOURCES += \
-    res/qml.qrc \
-    res/translations.qrc
+    res/qml.qrc 
+#    res/translations.qrc
 
 # osx target.
 osx: ICON = res/datovka.icns
@@ -417,23 +400,6 @@ ios {
     include(ios/ios.pri)
 }
 
-# android target.
-android {
-    contains(QMAKE_HOST.os, Windows) {
-        system(sed.exe -e "s/@VERSION@/$${VERSION}/g" "android/AndroidManifest.tmp" > "android/AndroidManifest.xml")
-    }
-    contains(QMAKE_HOST.os, Linux) {
-        system(sed -e "s/@VERSION@/$${VERSION}/g" "android/AndroidManifest.tmp" > "android/AndroidManifest.xml")
-    }
-    contains(QMAKE_HOST.os, Darwin) {
-        system(sed -e "s/@VERSION@/$${VERSION}/g" "android/AndroidManifest.tmp" > "android/AndroidManifest.xml")
-    }
-    ANDROID_PACKAGE_SOURCE_DIR = $$PWD/android
-    include(android/android.pri)
-
-    OTHER_FILES += android/src/cz/nic/mobiledatovka/java/QFileProvider.java
-}
-
 # winphone|winrt target.
 winphone|winrt: include(winrt/winrt.pri)
 
